### ISSUE TITLE
controller name rule not working?

### ISSUE THREAD

**Issue by johnpapa** [Gillespie59/eslint-plugin-angular#64]
This should have caught the bad controller name. It should require CustomersController.

```javascript
    angular
        .module('app.customers')
        .controller('customers', customers);

    /* @ngInject */
    function customers($state, dataservice, logger) {
```

**Comment by Gillespie59** [comment]
What is the configuration of the controller_name rule ?

**Comment by johnpapa** [comment]
```
env:
    node: true

plugins:
  - angular

rules:
    quotes: 2, "single"
    no-undef: 0
    no-use-before-define: 0
    no-unused-vars: 0
    key-spacing: 0
    no-shadow: 0
    wrap-iife: 0

    "angular/ng_controller_name": 2,
    "angular/ng_directive_name": 0,
    "angular/ng_empty_controller": 0,
    "angular/ng_filter_name": 0,
    "angular/ng_module_name": 0,
    "angular/ng_no_services": 2, ['$http', '$resource', 'Restangular']]
    "angular/ng_service_name": 0,
    "angular/ng_timeout_service": 2,
    "angular/ng_typecheck_array": 2,
    "angular/ng_typecheck_boolean": 2,
    "angular/ng_typecheck_date": 2,
    "angular/ng_typecheck_function": 2,
    "angular/ng_typecheck_number": 2,
    "angular/ng_typecheck_object": 2,
    "angular/ng_typecheck_regexp": 2,
    "angular/ng_typecheck_string": 2,
    "angular/ng_window_service": 2
```

**Comment by Gillespie59** [comment]
You should add a Regexp in the configuration of the rule : 

```
"angular/ng_controller_name": [2, /Controller$/]
```

I wanted to make this rule generic. Do you want this regexp as a default value ?

**Comment by johnpapa** [comment]
If it is going to follow the style guide it should be UpperCase and end with Controller by default.

**Comment by Gillespie59** [comment]
Ok perfect ;) Will do it

**Comment by johnpapa** [comment]
I guess that is the real question ... do you intend this plugin to follow the style guide?

**Comment by Gillespie59** [comment]
yes of course ;).

**Comment by johnpapa** [comment]
that's cool. i just wanted to make sure.

I talked with the angular team today about this and we're hoping others will help contribute so this can be run against an app to see how it follows the guide. If this gets fully developed, then it would be very helpful for those building angular 1.x apps and when it comes time to migrate

**Comment by Gillespie59** [comment]
:) I am very glad this plugin can help all NG developers !!! Thanks a lot for your feedbacks ! 

for the ng_controller_name rule, I will set the default value to /[A-Z].*Controller$/ . OK for you ?

**Comment by johnpapa** [comment]
Sounds right.



johnpapa 
@john_papa | http://johnpapa.net 

On March 6, 2015 at 2:59:04 AM, Emmanuel DEMEY (notifications@github.com) wrote:

:) I am very glad this plugin can help all NG developers !!! Thanks a lot for your feedbacks !

for the ng_controller_name rule, I will set the default value to /[A-Z].*Controller$/ . OK for you ?

**Comment by johnpapa** [comment]
thanks.

now you're locked in ... your face and name are in my slide deck for tomorrow now :)

**Comment by Gillespie59** [comment]
:) thanks. 
You should go to sleep now ?
